x-pack/legacy/plugins/code/server/routes/search.ts

```diff
@@ -38,9 +43,13 @@ export function repositorySearchRoute(router: CodeServerRouter, log: Logger) {
         page = parseInt(p as string, 10);
       }
 
-      let scope: string[] = [];
-      if (typeof repoScope === 'string') {
-        scope = repoScope.split(',');
+      const scope = await getScope(req, repoScope);
```

Comment:
nit: can we wrap this repeated code lines in to a function?

Reply:
I have refined the related code, please check if it is better.

Reply:
Thank you. It looks great!